lib/utils.ts

```diff
@@ -26,5 +26,55 @@ export const getNameString = (authors: string[] | string) => {
     return authors.join();
   }
 
-  return authors.slice(0, size - 1).join(', ') + ' and ' + authors[size - 1];
+  if (size <= 3) {
+    return authors.slice(0, size - 1).join(', ') + ' and ' + authors[size - 1];
+  }
+
+  return authors.slice(0, 2).join(', ') + ' and ' + (size - 2) + ' others';
+};
+
+function distance(lat1, lat2, lon1, lon2) {
+  lon1 = (lon1 * Math.PI) / 180;
+  lon2 = (lon2 * Math.PI) / 180;
+  lat1 = (lat1 * Math.PI) / 180;
+  lat2 = (lat2 * Math.PI) / 180;
+
+  let dlon = lon2 - lon1;
+  let dlat = lat2 - lat1;
+  let a =
+    Math.pow(Math.sin(dlat / 2), 2) +
+    Math.cos(lat1) * Math.cos(lat2) * Math.pow(Math.sin(dlon / 2), 2);
+
+  let c = 2 * Math.asin(Math.sqrt(a));
+
+  let r = 6371;
+
+  return c * r;
+}
+
+export const sortByDistance = (a, b) => {
+  const x0 = 41.56157392223945;
+  const y0 = -8.397397824887639;
+
+  const x1 = a.coordinates[0];
+  const y1 = a.coordinates[1];
+  const x2 = b.coordinates[0];
+  const y2 = b.coordinates[1];
+
+  const d1 = distance(x0, x1, y0, y1);
+  const d2 = distance(x0, x2, y0, y2);
+
+  return d1 - d2;
+};
+
+export const sortByOldest = (a, b) => {
+  return (
+    DateTime.fromISO(a.date).toMillis() - DateTime.fromISO(b.date).toMillis()
+  );
+};
+
+export const sortByLatest = (a, b) => {
+  return (
+    DateTime.fromISO(b.date).toMillis() - DateTime.fromISO(a.date).toMillis()
+  );
 };
```

Comment:
Don't you think it would make more sense to have this functions on `~/components/Sidebar/utils.ts` or even `~/components/Leaderboard/utils.ts`?

Reply:
Done!
Moved the functions to their respective `utils.ts` document. However, the ones that are used in more than one component (`getRelativeTimeString`, `getNameString` and `sortByOldest`) stayed on the same place.